.pfnci/linux/tests/_build_and_test.sh

```diff
@@ -9,13 +9,14 @@ pytest_opts=(
     --timeout 300
     --maxfail 500
     --showlocals
+    --numprocesses 2
```

Comment:
Could you rotate `CUDA_VISIBLE_DEVICES` for each worker process, like we do in chainer-test? This should prevent OOM on GPU and improve performance.

https://github.com/pytest-dev/pytest-xdist/issues/179#issuecomment-312419876

maybe something like this:

```import collections; q = collections.deque(range(2)); q.rotate(int(1)); print(",".join(map(str, q)))```

Max number of GPUs are given as environment variable "${CUPY_TEST_GPU_LIMIT}`

Reply:
I'll try `pytest_configure_node(node)` hook.
https://github.com/pytest-dev/pytest-xdist/blob/766e67ce526c88327670e9bcd4804c335df51015/src/xdist/newhooks.py#L38-L39

Reply:
I'm unsure if setting `CUDA_VISIBLE_DEVICES` after loading CUDA shared libraries will work, but seems worth trying.

Reply:
I think overriding `CUDA_VISIBLE_DEVICES` in test code is not always a good idea, because I sometimes want to run tests on different devices by manually setting `CUDA_VISIBLE_DEVICES`.

Reply:
I now think overriding `CUDA_VISIBLE_DEVICES` in test code makes sense as developers may want to use pytest-xdist interactively. 😄